### PR TITLE
nicotine-plus 2.1.2 (new formula)

### PR DESCRIPTION
**Pull request by EricFromCanada**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----
Replacement for the Python 2-only formula deleted in #47748.

### CODE REVIEW

File: Formula/nicotine-plus.rb [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+  desc "Graphical client for the SoulSeek peer-to-peer system"
+  homepage "https://www.nicotine-plus.org/"
+  url "https://github.com/Nicotine-Plus/nicotine-plus/archive/2.1.2.tar.gz"
+  sha256 "3ed18ade97183c632836eb8e304a515fc19a35babb46cc6e6747bcfd8205dcdf"
+  license "GPL-3.0-or-later"
+  head "https://github.com/Nicotine-Plus/nicotine-plus.git"
+
+  depends_on "adwaita-icon-theme"
+  depends_on "geoip"
```
Comment (mathiascode) [inline Formula/nicotine-plus.rb:12]: geoip is no longer used in Nicotine+.

**Review by chenrui333 — DISMISSED** [review]
lgtm

File: Formula/nicotine-plus.rb [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+  homepage "https://www.nicotine-plus.org/"
+  url "https://github.com/Nicotine-Plus/nicotine-plus/archive/2.1.2.tar.gz"
+  sha256 "3ed18ade97183c632836eb8e304a515fc19a35babb46cc6e6747bcfd8205dcdf"
+  license "GPL-3.0-or-later"
+  head "https://github.com/Nicotine-Plus/nicotine-plus.git"
+
+  depends_on "adwaita-icon-theme"
+  depends_on "gtk+3"
+  depends_on "pygobject3"
+  depends_on "python@3.8"
```
Comment (dawidd6) [inline Formula/nicotine-plus.rb:14]: Doesn't work with python@3.9?

Comment (EricFromCanada) [reply Formula/nicotine-plus.rb]: It does, but that causes CI to fail:
```
  * nicotine-plus contains conflicting version recursive dependencies:
      python@3.8, python@3.9
    View these with `brew deps --tree nicotine-plus`.
```

**Review by SMillerDev — APPROVED** [review]
Thanks @EricFromCanada  ! Without your contributions it'd be impossible to keep homebrew going. You can feel good knowing that you've made the world a lot better for homebrew users around the world! 👍 🎉

**Pull request closed without merging**